types/vue.d.ts

```diff
@@ -67,8 +68,51 @@ export declare class Vue {
   $nextTick(callback: (this: this) => void): void;
   $nextTick(): Promise<void>;
   $createElement: CreateElement;
+}
+
+export type CombinedVueInstance<Data, Methods, Computed, Props> = Data & Methods & Computed & Props & Vue;
+export type ExtendedVue<Constructor extends VueConstructor, Data, Methods, Computed, Props> =
+  (new (...args: any[]) => CombinedVueInstance<Data, Methods, Computed, Props>) &
```

Comment:
> As ExtendedVue is a sub-class of the Vue constructor, we can expect the constructor signature would only receive one argument like below?

@ktsn ExtendedVue creates an intersection with the mixin constructor `(new (...args: any[]) => CombinedVueInstance<Data, Methods, Computed, Props>)`. This means it just absorbs the argument types of the `Constructor` type that's given. See https://github.com/Microsoft/TypeScript/pull/13743 for more details